qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -280,6 +280,21 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return self._run(dag, tracker)
 
     def _run(self, dag: DAGCircuit, tracker: QubitTracker) -> DAGCircuit:
+        # Check if HighLevelSynthesis can be skipped.
+        all_skipped = True
+        for node in dag.topological_op_nodes():
+            qubits = tuple(dag.find_bit(q).index for q in node.qargs)
+            if not (
+                dag.has_calibration_for(node)
+                or len(node.qargs) < self._min_qubits
+                or node.is_directive()
+                or self._definitely_skip_node(node, qubits)
+            ):
+                all_skipped = False
+                break
+        if all_skipped:
+            return dag
```

Comment:
You don't have to do this, but in case you're interested, Python has a very uncommon control-flow construction that handles this precise case without an auxiliary variable:
```python
for node in dag.topological_op_nodes():
    qubits = ...
    if not self._should_skip(dag, node, qubits):
        break
else:
    return dag
```

(The `else` is attached to the `for` loop.)  Totally up to you if you want to use it - there's plenty of reason not to, since most people don't know what the `for/else` pattern in Python means.

Reply:
This is neat, I like this!

Reply:
Done in [0962be4](https://github.com/Qiskit/qiskit/pull/13070/commits/0962be43f9f3fef986f04aa62a7c3a3ba6b955d6).